docs/headerMiddleware.md

```diff
@@ -0,0 +1,171 @@
+| `dataRules` | `{Object}` | [Model transformation rules](http://docs.fluidproject.org/infusion/development/ModelTransformationAPI.html) that control what data will be combined with `template` (see below) |
+
+The simplest header definition looks something like the following
+
+    headers: {
+        static: {
+            fieldName: "Access-Control-Allow-Origin",
+            template:   "*"
+        }
+    }
+
+Note that the template has no variables (which would look something like `%var`).  As a result, the literal value `*` is
+set.  In this case, no data is interpolated, so `dataRules` is ignored and can be safely omitted.
+
+A full definition looks something like the following:
+
+    headers: {
+        dynamic: {
+            fieldName: "Content-Type",
+            template: "%var",
+            dataRules: {
+                var: "request.query.var"
+            }
+        }
+    }
+
+If we were accessing a URL like `http://my.site.com/path/to/router?var=application%2Fjson`, this component would set
+the `Content-Type` header to `application/json`.
+
+The rules defined in `dataRules` have access to `that`, the component itself, and `request`, the
+[Express request object](expressjs.com/en/api.html#req).  In the example above, we have pulled query data from the
+`request` object.  We could just has easily have read a model variable from our component, or an option from our
+component.  Here is an example illustrating a few possibilities:
+
+    headers: {
+        deleteCookie: {
+            fieldName: "Set-Cookie",
+            template:  "existingCookie=deleted; Expires=Thu, 01-Jan-1970 00:00:01 GMT"
+        },
+        setModelCookie: {
+            fieldName: "Set-Cookie",
+            template:  "%cookieName=%cookieValue",
+            dataRules: {
+                cookieName:  "that.model.cookie.name",
+                cookieValue: "that.model.cookie.value"
+            }
+        },
+        refresh: {
+            fieldName: "Refresh",
+            template: "%seconds; url=%url",
+            dataRules: {
+                seconds: { literalValue: 5 },
+                url:     "that.options.refreshUrl"
+            }
+        }
+    }
+
+With those rules, this component would:
+
+1. Set the `Set-Coookie` header to `{that.model.cookie.name}={that.model.cookie.value}`
+2. Set the `Refresh` header to `5; url={that.options.refreshUrl}`
+
+Note that the hard-coded value of `5` in the `refresh` example is set in a `dataRules` block using `literalValue`.  You
+can also set static values in the `template` value itself, as illustrated in the `deleteCookie` example.
+
+## Invokers
+
+### `{middleware}.middleware(req, res, next)`
+
+* `req`: The [request object](http://expressjs.com/en/api.html#req) provided by Express, which wraps node's [`http.incomingMessage`](https://nodejs.org/api/http.html#http_class_http_incomingmessage).
```

Comment:
You are correct.  I missed it in my previous cull because the variable names were also different.  I have searched throughout for both variations, this was the only holdout, and is fixed.